### PR TITLE
Fix Gradle plugin debug mode and improve documentation

### LINKED ISSUE #2653: The option "--debug=true" of quarkusDev is catched by gradle
**Can't use --debug=true with gradle to activate suspend=y**

As the documentation states this command should be fine:
```
quarkus-quickstarts-0.15.0/getting-started $ ./gradlew quarkusDev  --debug=true
```
but with it gradle complains with this message:
```
Command-line option '--debug' does not take an argument.
```
The option is indeed catched by gradle instead of been passed to QuarkusDev.
**To reproduce**:  
I've used the *quarkus quick starts* so it can be used also to reproduce the problem.

### PR DESCRIPTION
**Pull request by gwenneg**
Resolves #2653

### CODE REVIEW

**Comment by gwenneg** [comment]
Since there's no way to use the `--debug` Gradle option for Quarkus, I replaced it with a simple system property. The Gradle and Maven debugging parameters now look exactly the same.

This comes with some documentation improvements which should probably be done in the Maven tooling guide (assuming the implementations are similar, I didn't have time to confirm that). Depending on the review, I might submit another PR to improve the Maven tooling guide (or do it in this one if you prefer).

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
 
     private Set<File> filesIncludedInClasspath = new HashSet<>();
 
-    private String debug;
+    private String debug = System.getProperty("debug");
```
Comment (feaster83) [inline devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java]: This line can be removed. The `startDev` method is the only method that use/read the debug flag so it's better to move this code line to that method and read in into a method local variable debug. Place it right before the if statements.

**Review by feaster83 — CHANGES_REQUESTED** [review]
Can be approved after the comment on line 66 is discussed/resolved (see comment)

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java (hunk 1 above)
Comment (gwenneg) [reply devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java]: You're right, thanks! That's fixed now.

**Review by feaster83 — APPROVED** [review]

**Comment by geoand** [comment]
So IIUC with this change one can't simply do `-Ddebug` like they can in Maven, right?

**Comment by gwenneg** [comment]
@geoand Well, with the current quarkus-gradle-plugin behavior (which I didn't change in this PR), there's no point running a simple `-Ddebug` because the default behavior already consists in running the server in debug bug.

This code throws a `org.gradle.api.GradleException: Invalid value for debug parameter:  must be true|false|client|{port}` if you run `./gradlew quarkusDev -Ddebug`. This can be changed to just keep the default behavior for this command but I don't think it's worth mentionning in the doc.

**Comment by geoand** [comment]
What I am saying is that when you run the Maven Plugin with `-Ddebug` , then suspend is enabled. Shouldn't we do the same for the Gradle plugin?

**Comment by gwenneg** [comment]
Are you sure it's suspended with a `-Ddebug` using Maven? From what I can see [here](https://github.com/quarkusio/quarkus/blob/master/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java#L189), it shouldn't be.

**Comment by gwenneg** [comment]
Now that I had a look at the Maven plugin code, it looks like the Quarkus doc about the Maven plugin needs the same kind of changes I did for the Gradle plugin.

**Comment by geoand** [comment]
Yes, I am certain that `-Ddebug` suspends, I use it all the time 😛

**Comment by gwenneg** [comment]
I just tested all `-Ddebug` possibe values with the Maven plugin, running `-Ddebug` or `-Ddebug=true` produce the same effect which makes `-Ddebug=true` useless.

I'm not sure this was done on purpose but I could change the PR to have the same behavior with the Gradle plugin.

**Comment by geoand** [comment]
I personally find `-Ddebug` very useful and I am pretty sure others use it as well.

**Comment by gwenneg** [comment]
When I said I wasn't sure the current behavior was done on purpose, that's because of this sentence in the Maven plugin javadoc:
> The default is to start in debug mode without suspending and listen on port 5005.

I will change this PR to make `-Ddebug` and `-Ddebug=true` do the same thing with the Gradle plugin and maybe we'll discuss wether or not the server should be suspended using `-Ddebug` with both plugins in another issue.

**Comment by stuartwdouglas** [comment]
The current behaviour is definitely intentional. -Ddebug is shorthand for -Ddebug=true, which is standard for boolean maven configuration.

**Comment by gwenneg** [comment]
Thanks for this info @stuartwdouglas! I've updated the PR to use the word `shorthand` in the doc.

**Review by geoand — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**